tests/xargs_test.py

```diff
@@ -20,30 +21,30 @@ def test_cpu_count_sched_getaffinity_exists():
         assert xargs.cpu_count() == 345
 
 
-@pytest.fixture
+@contextlib.contextmanager
 def no_sched_getaffinity():
     # Simulates an OS without os.sched_getaffinity available (mac/windows)
     # https://docs.python.org/3/library/os.html#interface-to-the-scheduler
-    with mock.patch.object(
-            os,
-            'sched_getaffinity',
-            create=True,
-            side_effect=AttributeError,
-    ):
+    if hasattr(os, 'sched_getaffinity'):  # pragma: no cover (win32 py38)
+        func = os.sched_getaffinity
+        del os.sched_getaffinity
         yield
+        os.sched_getaffinity = func
+    else:
+        yield  # pragma: no cover (darwin)
```

Comment:
you're looking for `# pragma: posix cover` and `# pragma: posix no cover` which uses `covdefaults` to mark these appropriately

Reply:
TIL these tokens have semantic meaning

Reply:
TIL

Reply:
it would need try + finally

Reply:
Oh I meant TIL that purest fixtures are exception-safe. I realize without try/finally my code isn't exception-safe